### PR TITLE
Directory search by tag and location

### PR DESCRIPTION
**Pull request by laurenho025**
## Description
Hooked up directory search to include tag and location. Unsure if search by location is working correctly / if I updated the api wrapper right for search by tag? @alicesf2 @josh-byster if you could check that for me in case I did it wrong

Fixes #<number>

### CODE REVIEW

File: frontend/src/utils/api.js [hunk 1]
```diff
@@ -4,16 +4,24 @@ import {
...
   const defaultRadius = 500;
   let endptStr = `resources/filter?`;
 
-  if (address && keyword) {
+  if (address && keyword && tag) {
+    endptStr += `radius=${defaultRadius}&address=${address}&keyword=${keyword}&tag=${tag}`;
+  } else if (address && keyword) {
     endptStr += `radius=${defaultRadius}&address=${address}&keyword=${keyword}`;
+  } else if (address && tag) {
+    endptStr += `radius=${defaultRadius}&address=${address}&tag=${tag}`;
```
Comment (alan-fang) [inline frontend/src/utils/api.js]: let's do something like:
```
if (radius) {
    endptStr  += `radius=${defaultRadius}`
}
if (address) {
    endptStr += `&address=${address}&keyword=${keyword}`;
  }
if (keyword) {
    endptStr += `&keyword=${keyword}`;
  } 
if (tag) {
    endptStr += `&tag=${tag}`;
}
```

so we don't have to enumerate all possibilities

Comment (alicesf2) [reply frontend/src/utils/api.js]: wait but if there's no radius there would be an extra '&' at the beginning of the other fields

Comment (alan-fang) [reply frontend/src/utils/api.js]: radius is always set, this will just prep us for being able to pass in radius as a search param too

**Comment by laurenho025** [comment]
>             ∩
> 　　　　　　　＼＼
> 　　　　　　　／　 ）
> ⊂＼＿／￣￣￣　 ／
> 　＼＿／   ° ͜ʖ ° （
> 　　　）　　 　／⌒＼
> 　　／　 ＿＿＿／⌒＼⊃
> 　（　 ／
> 　　＼＼
>      U

I don't like this

File: frontend/src/utils/api.js (hunk 1 above)
Comment (josh-byster) [reply frontend/src/utils/api.js]: We can actually just do this really cleanly using `urljoin` which is a module we're using for the Redux API middleware. You can import from `url-join` like in `apiHelpers.js` and then do something like [this](https://github.com/jfromaniello/url-join/blob/master/test/tests.js#L83). To be really fancy you can try using the [spread operator](https://medium.com/@luke_smaki/javascript-es6-spread-operator-and-rest-parameters-b3e89d112281) to put this in to parameters in the function.

File: frontend/src/pages/DirectoryView/SearchBar/index.jsx [hunk 2]
```diff
@@ -4,18 +4,43 @@ import { Button } from "reactstrap";
...
+  };
+
   render() {
     return (
       <div className="searchbar-wrapper">
-        <form className="search">
+        <form className="search" onSubmit={this.handleSubmit}>
           <label>
-            <input id="search-general" type="text" />
+            <input
```
Comment (josh-byster) [inline frontend/src/pages/DirectoryView/SearchBar/index.jsx]: Nice work here using `onChange` handlers. One thing that we want to ensure when we do this is that we're doing a fully "Controlled Component". What this means is that the input fields receive as props the value and when you change the field, you call the onChange handler (like you do here). What this does is allows us to not have two pieces of data that can somehow become out of sync (the parent's recordd of what's in this textbox which we hold in state and what's actually in the textbox). See [Controlled Components.](https://reactjs.org/docs/forms.html)

File: frontend/src/utils/api.js (hunk 1 above)
Comment (alan-fang) [reply frontend/src/utils/api.js]: ah, that's what i was looking for!

**Pull request merged**